lib/facility_factory.rb

```diff
@@ -0,0 +1,17 @@
+class FacilityFactory
+  attr_accessor :facility_instances
+
+  def initialize
+    @facility_instances = []
+  end
+
+  def create_facilities(facilities_data)
+    @facility_instances = facilities_data.map do |facility|
+      Facility.new(
+        name: facility[:dmv_office] || facility[:office_name] || facility[:name],
+        address: facility[:address_li] || facility[:street_address_line_1] || facility[:address1],
+        phone: facility[:phone] || facility[:public_phone_number] || facility[:phone]
+      )
+    end
+  end
+end
```

Comment:
I don't think you need the `:facility_instances` method ... or the instance variable. We really only need the `create_facilities` method to return a collection of Facility objects. No need to store state, since this response will change anytime the data passed in is changed.

Reply:
I know you asked for feedback specifically on lines 8-14. THis approach looks good to me! It should work for this set of data othat you're working with. It's possible that once another set of data is added for another state, you might need to refactor this, but for now, looks great!